internal/operator-controller/applier/boxcutter.go

```diff
@@ -397,7 +397,14 @@ func (bc *Boxcutter) apply(ctx context.Context, contentFS fs.FS, ext *ocv1.Clust
 	if progressingCondition == nil && availableCondition == nil && succeededCondition == nil {
 		return false, "New revision created", nil
 	} else if progressingCondition != nil && progressingCondition.Status == metav1.ConditionTrue {
-		return false, progressingCondition.Message, nil
+		switch progressingCondition.Reason {
+		case ocv1.ReasonSucceeded:
+			return true, "", nil
+		case ocv1.ClusterExtensionRevisionReasonRetrying:
+			return false, "", errors.New(progressingCondition.Message)
+		default:
+			return false, progressingCondition.Message, nil
+		}
 	} else if availableCondition != nil && availableCondition.Status != metav1.ConditionTrue {
 		return false, "", errors.New(availableCondition.Message)
 	} else if succeededCondition != nil && succeededCondition.Status != metav1.ConditionTrue {
```

Comment:
[nit] would it possible to simplify/make more readable this big `if-else-swtich-else-else` block?